src/components/ItemList.svelte

```diff
@@ -28,6 +29,47 @@
   let topElement;
   let scrollY;
 
+  const searchIndex = new Document({
+    tokenize: "forward",
+    index: ["id", "type", "tags", "origin", "description"],
+  });
+
+  items.forEach((item) => {
+    searchIndex.add({
+      id: item.name,
+      type: item.type,
+      tags: item.tags,
+      origin: item.origin,
+      description: item.description,
+    });
+  });
+
+  function fullTextSearch(query, searchItems) {
+    let results = [];
+    // combine results of all fields
+    searchIndex.search(query).forEach((result) => {
+      results = [...results, ...result.result];
+    });
+
+    return results.map((result) => {
+      return searchItems.find((item) => item.name === result);
+    });
```

Comment:
I think we can simplify this a bit.

```suggestion
    // combine results of all fields
    return searchIndex.search(query).map((result) => {
      return searchItems.find((item) => item.name === result.result);
    });
```

Reply:
I don't think we can simplify this. An example of what `searchIndex.search(query)` returns: 

```
[
  {
    "field": "id",
    "result": [
      "about_page.libraries_tapped",
      "about_page.licensing_tapped",
      "about_page.privacy_notice_tapped",
      "about_page.rights_tapped",
      "about_page.support_tapped"
    ]
  },
  {
    "field": "description",
    "result": [
      "metrics.start_reason_process_error",
      "about_page.licensing_tapped",
      "about_page.privacy_notice_tapped",
      "about_page.support_tapped",
      "geckoview.page_reload_time",
      "about_page.libraries_tapped",
      "about_page.rights_tapped",
      "perf.startup.home_activity_on_start"
    ]
  }
]
```
Since there are duplicates in `result` array for each field, we need to first combine and deduplicate the results: `results = [...results, ...result.result]`. Then return only the items in `searchItems` that match the item names in `results`. Maybe there's a different way to simplify this that I'm not seeing?

Reply:
Oh, right! Sorry, I misread your code. I still think we can use [flat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) here to concatenate the results, then uniquify the result using Set:

https://stackoverflow.com/questions/9229645/remove-duplicate-values-from-js-array

Reply:
thanks! I used set to uniqfy the result in the latest commit. Could you explain why using flat is better than the spread operator in this case?

Reply:
I think you forgot to push your commit. 😄 

Wrt your question: this is a little subjective, but in general I think it's slightly better practice to use map() instead of forEach() + an accumulator (the spread operator in this case). It's a little closer to a pure functional style, which (all things being equal) tends to be easier understand and less prone to bugs. I like this stackoverflow answer on the subject: https://stackoverflow.com/a/43883957/295132

It occurs to me now that maybe you could use flatMap here, which might be even more elegant as well as (very slightly) more efficient:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap

Reply:
These are great! Thank you @wlach!